storage/innobase/trx/trx0rseg.cc

```diff
@@ -106,27 +104,33 @@ void
 trx_rseg_update_wsrep_checkpoint(
 	buf_block_t*	rseg_header,
 	const XID*	xid,
-	mtr_t*		mtr)
+	mtr_t*		mtr,
+	bool for_rollback)
 {
 	ut_ad(wsrep_is_wsrep_xid(xid));
 
-#ifdef UNIV_DEBUG
 	/* Check that seqno is monotonically increasing */
 	long long xid_seqno = wsrep_xid_seqno(xid);
 	const byte* xid_uuid = wsrep_xid_uuid(xid);
 
 	if (xid_seqno != -1
 	    && !memcmp(xid_uuid, wsrep_uuid, sizeof wsrep_uuid)) {
-		ut_ad(xid_seqno > wsrep_seqno);
+		/* a safety measure, make sure that certification failure
+		does not record a checkpoint too far ahead, this would
+		show that the node has committed seqno wsrep_seqno+1,
+		although it is still processing */
+		if (for_rollback && xid_seqno > wsrep_seqno + 1) {
```

Comment:
This will allow syncing `xid_seqno` which is lower than seqno which has already been synced, which may cause recovery to wrong position.